test/operations.py

```diff
@@ -0,0 +1,61 @@
+import requests
+import json
+from typing import Optional, Tuple
+
+
+def create_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.put(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def check_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.get(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def delete_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.delete(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def delete_document(endpoint: str, index_name: str, doc_id: str, auth: Optional[Tuple[str, str]] = None):
+    response = requests.delete(f'{endpoint}/{index_name}/_doc/{doc_id}', auth=auth, verify=False)
+
+    return response
+
+
+def create_document(endpoint: str, index_name: str, doc_id: str, auth: Optional[Tuple[str, str]] = None,
+                    data: Optional[dict] = None):
+    document = {
+        'title': 'Test Document',
+        'content': 'This is a sample document for testing OpenSearch.'
+    }
+    url = f'{endpoint}/{index_name}/_doc/{doc_id}'
+    headers = {'Content-Type': 'application/json'}
+
+    response = requests.put(url, headers=headers, data=json.dumps(document), auth=auth, verify=False)
+
+    return response
+
+
+def check_document(endpoint, index_name, doc_id, auth=None):
```

Comment:
minor: would "retrieve_document" be a better name for this? "check" sounds to me like it's going to return true/false about whether the document exists, but this just returns either the document itself or a 404, correct?

Reply:
Also, I think you may have missed the type hints for this function